packages/desktop-client/src/components/mobile/transactions/TransactionListWithBalances.tsx

```diff
@@ -0,0 +1,278 @@
+    >
+      <InputWithContent
+        leftContent={
+          <SvgSearchAlternate
+            style={{
+              width: 13,
+              height: 13,
+              flexShrink: 0,
+              color: text ? theme.formInputTextHighlight : 'inherit',
+              margin: 5,
+              marginRight: 0,
+            }}
+          />
+        }
+        value={text}
+        onChangeValue={text => {
+          setText(text);
+          onSearch(text);
+        }}
+        placeholder={placeholder}
+        style={{
+          backgroundColor: theme.tableBackground,
+          border: `1px solid ${theme.formInputBorder}`,
+          flex: 1,
+          height: styles.mobileMinHeight,
+        }}
+      />
+    </View>
+  );
+}
+
+type TransactionListWithBalancesProps = {
+  isLoading: boolean;
+  transactions: Readonly<TransactionEntity[]>;
+  balance:
+    | ReturnType<typeof queries.onBudgetAccountBalance>
+    | ReturnType<typeof queries.offBudgetAccountBalance>
+    | ReturnType<typeof queries.uncategorizedBalance>
+    | ReturnType<typeof queries.categoryBalance>
+    | ReturnType<typeof queries.accountBalance>
+    | ReturnType<typeof queries.allAccountBalance>;
+  balanceCleared?:
+    | ReturnType<typeof queries.categoryBalanceCleared>
+    | ReturnType<typeof queries.accountBalanceCleared>;
+  balanceUncleared?:
+    | ReturnType<typeof queries.categoryBalanceUncleared>
+    | ReturnType<typeof queries.accountBalanceUncleared>;
+  searchPlaceholder: string;
+  onSearch: (searchText: string) => void;
+  isLoadingMore: boolean;
+  onLoadMore: () => void;
+  onOpenTransaction: (transaction: TransactionEntity) => void;
+  onRefresh?: () => void;
+};
+
+export function TransactionListWithBalances({
+  isLoading,
+  transactions,
+  balance,
+  balanceCleared,
+  balanceUncleared,
+  searchPlaceholder = 'Search...',
+  onSearch,
+  isLoadingMore,
+  onLoadMore,
+  onOpenTransaction,
+  onRefresh,
+}: TransactionListWithBalancesProps) {
+  const selectedInst = useSelected('transactions', [...transactions], []);
+
+  return (
+    <SelectedProvider instance={selectedInst}>
+      <>
+        <View
+          style={{
+            flexShrink: 0,
+            marginTop: 10,
+          }}
+        >
+          <View
+            style={{
+              flexDirection: 'row',
+              justifyContent: 'space-evenly',
+            }}
+          >
+            {balanceCleared && balanceUncleared ? (
+              <BalanceWithCleared
+                balance={balance}
+                balanceCleared={balanceCleared}
+                balanceUncleared={balanceUncleared}
+              />
+            ) : (
+              <Balance balance={balance} />
+            )}
+          </View>
+          <TransactionSearchInput
+            placeholder={searchPlaceholder}
+            onSearch={onSearch}
+          />
+        </View>
+        <PullToRefresh
+          isPullable={!!onRefresh}
+          onRefresh={async () => onRefresh?.()}
```

Comment:
this props accept an async [function](https://github.com/thmsgbrt/react-simple-pull-to-refresh/blob/b91b3eff46210b5c5e7e95d6b02a83e0d1a4eb45/README.md?plain=1#L72)